xcpmd/src/acpi-events.c

```diff
@@ -30,133 +30,403 @@
 #include "project.h"
 #include "xcpmd.h"
 #include "acpi-events.h"
+#include "battery.h"
 
 static int acpi_events_fd = -1;
 static struct event acpi_event;
 
-static void write_state_info_in_xenstore(FILE *file, char *xenstore_path,
-             char *search_str, char *default_value, char *alternate_value)
-{
-    char file_data[1024];
 
-    if ( file == NULL )
-        return;
-
-    xenstore_write(default_value, xenstore_path);
+void adjust_brightness(int increase, int force) {
 
-    memset(file_data, 0, 1024);
-    fgets(file_data, 1024, file);
-    if (strstr(file_data, search_str))
-        xenstore_write(alternate_value, xenstore_path);
+    if ( force || (pm_quirks & PM_QUIRK_SW_ASSIST_BCL) || (pm_quirks & PM_QUIRK_HP_HOTKEY_INPUT)) {
+        if (increase)
+            com_citrix_xenclient_surfman_increase_brightness_(xcdbus_conn, SURFMAN_SERVICE, SURFMAN_PATH);
+        else
+            com_citrix_xenclient_surfman_decrease_brightness_(xcdbus_conn, SURFMAN_SERVICE, SURFMAN_PATH);
+    }
 }
 
-void initialize_system_state_info(void)
-{
+
+int get_ac_adapter_status(void) {
+
+    char data[128];
     FILE *file;
+    file = fopen(AC_ADAPTER_STATE_FILE_PATH, "r");
 
-    file = get_ac_adpater_state_file();
-    write_state_info_in_xenstore(file,
-                                XS_AC_ADAPTER_STATE_PATH, "0", "1", "0");
-    if ( file != NULL )
-        fclose(file);
+    //If the file doesn't exist, we're not a laptop.
+    if (file == NULL && errno == ENOENT)
+        return NO_AC;
+    else if (file == NULL)
+        return AC_UNKNOWN;
+
+    fgets(data, sizeof(data), file);
```

Comment:
are you guaranteed to get all of the data desired on fgets? and be null terminated?